src/main/savestates.c

```diff
@@ -1477,7 +1484,13 @@ static void savestates_save_m64p_work(struct work_struct *work)
     SDL_LockMutex(savestates_lock);
 
     // Write the state to a GZIP file
+#ifdef _WIN32
+    wchar_t wstr_filepath[PATH_MAX];
+    MultiByteToWideChar(CP_UTF8, 0, save->filepath, -1, wstr_filepath, PATH_MAX);
+    f = gzopen_w(wstr_filepath, "wb");
+#else
     f = gzopen(save->filepath, "wb");
+#endif
```

Comment:
Same here: Does it make sense to have a `osal_gzopen()` in `osal/files.c`?

@richard42